lib/promscrape/discovery/azure/machine.go

```diff
@@ -78,7 +80,8 @@ func visitAllAPIObjects(ac *apiConfig, apiURL string, cb func(data json.RawMessa
 				return err
 			}
 		}
-		nextLink = lar.NextLink
+		// Azure API returns nextLink with apiServer in it, so we need to remove it
```

Comment:
I suggest to change semantic, rename nextLink for nextLinkURI, make for loop without condition and check nextLink at the end of the loop:
```go
               if lar.NextLink == "" {
			break
		}
		nextURL, err := url.Parse(lar.NextLink)
		if err != nil {
			return err
		}
		nextLinkURI = nextURL.RequestURI()
```

Replace look unreliable for me, since it may omit scheme or port at nextLink.

Reply:
Implemented this approach, thanks for suggestion!